server/src/job.rs

```diff
@@ -0,0 +1,74 @@
+use std::os::unix::process::ExitStatusExt;
+use std::process::{ExitStatus, Stdio};
+use std::sync::Arc;
+use tokio::process::{Child, Command};
+
+use common::job_status::Completed;
+use common::output_event::Stream as OutputStream;
+use common::*;
+
+use crate::client_cert::ClientName;
+use crate::output_stream::OutputHandler;
+
+/// Map status of a completed process to `JobStatus`
+fn completed_status(status: ExitStatus) -> JobStatus {
+    if let Some(value) = status.code() {
+        JobStatus {
+            completed: Some(Completed::StatusCode(value)),
+        }
+    } else if let Some(value) = status.signal() {
+        JobStatus {
+            completed: Some(Completed::Signal(value)),
+        }
+    } else {
+        panic!("Unknown process exit state")
+    }
+}
+
+/// A single running job, i.e. a process
+pub struct Job {
+    pub owner: ClientName,
+    child: Child,
+    pub stdout: Arc<OutputHandler>,
+    pub stderr: Arc<OutputHandler>,
+}
+impl Job {
+    pub fn spawn(owner: ClientName, req: JobStartRequest) -> Result<Self, String> {
```

Comment:
You're right, this will cause zombie processes. I though that Tokio should be able to handle this by itself, as the documentation says

> The tokio runtime will, on a best-effort basis, attempt to reap and clean up any process which it has spawned. No additional guarantees are made with regards how quickly or how often this procedure will take place.

Looking more into it, this only concerns cases when the  `Child` handle has been dropped. Since this is not the case, the best-effort orphan reaper will not run.

I've fixed this in by 12782b1 managing the child process in a separate task.